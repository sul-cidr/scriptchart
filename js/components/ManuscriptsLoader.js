import React from "react";

/* Mock data to use in development when local API backend is not available */
const fallbackManuscripts = [
  { id: 20, shelfmark: "BL. Add. 12146", date: "1007 " },
  { id: 21, shelfmark: "BL. Or. 8732", date: "770" },
  { id: 22, shelfmark: "BL. Or. 8606", date: "723" },
  { id: 23, shelfmark: "BL. Add. 17102", date: "598-599" },
  { id: 24, shelfmark: "BL. Add. 17107", date: "540-541" },
  { id: 25, shelfmark: "BL. Add. 12175", date: "533-534 " },
  { id: 26, shelfmark: "BL. Add. 17109", date: "873 - 874" }
];

class ManuscriptsLoader extends React.Component {
  constructor(props) {
    super(props);
    this.state = { manuscripts: [] };
  }
  componentDidMount() {
    fetch("http://localhost:8000/api/manuscripts?format=json")
      .then(response => {
        return response.json();
      })
      /* In production, it's likely preferable for the menu to display
       * a blank list when the backend API is down, rather than
       * displaying mock data that does not reflect the database state.
       */
      .catch(function(error) {
        return fallbackManuscripts;
      })
      .then(data => {
        let ms = data.map(manuscript => {
          return <option key={manuscript.id}>{manuscript.shelfmark}</option>;
        });
        this.setState({ manuscripts: ms });
      });
  }
  render() {
    return <select multiple>{this.state.manuscripts}</select>;
  }
}

export default ManuscriptsLoader;
