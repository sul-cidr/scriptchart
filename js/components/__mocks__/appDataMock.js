export const tableData = {"9":{"2":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Alaph (Angular)_611_277_75_50.png","id":44407,"top":277,"left":611,"width":75,"height":50},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Alaph (Angular)_557_289_70_48.png","id":44412,"top":289,"left":557,"width":70,"height":48},{"page":1481,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-28a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_28a_Alaph (Angular)_1016_249_64_44.png","id":44408,"top":249,"left":1016,"width":64,"height":44},{"page":1481,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-28a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_28a_Alaph (Angular)_727_357_63_44.png","id":44409,"top":357,"left":727,"width":63,"height":44},{"page":2902,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-11b.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_11b_Alaph (Angular)_252_393_55_46.png","id":44402,"top":393,"left":252,"width":55,"height":46}],"19":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Kaph_304_277_44_39.png","id":44503,"top":277,"left":304,"width":44,"height":39},{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Kaph_727_342_41_40.png","id":44504,"top":342,"left":727,"width":41,"height":40},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Kaph_1034_350_43_44.png","id":44507,"top":350,"left":1034,"width":43,"height":44},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Kaph_959_463_42_38.png","id":44508,"top":463,"left":959,"width":42,"height":38},{"page":1156,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-13b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_13b_Kaph_504_762_42_38.png","id":44501,"top":762,"left":504,"width":42,"height":38}],"37":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Sadhe_732_1071_106_59.png","id":44593,"top":1071,"left":732,"width":106,"height":59},{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Sadhe_465_1427_75_54.png","id":44594,"top":1427,"left":465,"width":75,"height":54},{"page":1156,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-13b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_13b_Sadhe_582_1301_76_51.png","id":44592,"top":1301,"left":582,"width":76,"height":51},{"page":1377,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr112/VatSyr112-006.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_3L_Sadhe_735_855_61_46.png","id":9863,"top":855,"left":735,"width":61,"height":46},{"page":1377,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr112/VatSyr112-006.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_3L_Sadhe_636_1134_62_50.png","id":9864,"top":1134,"left":636,"width":62,"height":50}]},"11":{"2":[{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_201_258_23_24.png","id":74759,"top":258,"left":201,"width":23,"height":24},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_278_254_22_24.png","id":74760,"top":254,"left":278,"width":22,"height":24},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_125_281_28_25.png","id":74761,"top":281,"left":125,"width":28,"height":25},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_212_283_22_20.png","id":74762,"top":283,"left":212,"width":22,"height":20},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_237_278_22_25.png","id":74763,"top":278,"left":237,"width":22,"height":25}],"19":[{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_675_615_15_15.png","id":13256,"top":615,"left":675,"width":15,"height":15},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_299_214_18_13.png","id":13294,"top":214,"left":299,"width":18,"height":13},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_608_731_16_15.png","id":13346,"top":731,"left":608,"width":16,"height":15},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Kaph_289_518_18_13.png","id":10180,"top":518,"left":289,"width":18,"height":13},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Kaph_585_808_15_13.png","id":10181,"top":808,"left":585,"width":15,"height":13}],"37":[{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_432_234_21_21.png","id":13238,"top":234,"left":432,"width":21,"height":21},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_420_262_24_19.png","id":13239,"top":262,"left":420,"width":24,"height":19},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_624_417_23_21.png","id":13248,"top":417,"left":624,"width":23,"height":21},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_647_296_26_21.png","id":13259,"top":296,"left":647,"width":26,"height":21},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Sadhe_760_293_26_17.png","id":10253,"top":293,"left":760,"width":26,"height":17}]},"16":{"2":[{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_759_1756_86_99.png","id":70463,"top":1756,"left":759,"width":86,"height":99},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_636_1925_95_93.png","id":70464,"top":1925,"left":636,"width":95,"height":93},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_667_1133_68_75.png","id":70465,"top":1133,"left":667,"width":68,"height":75},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_1377_1609_72_77.png","id":70466,"top":1609,"left":1377,"width":72,"height":77},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_1721_1791_86_82.png","id":70467,"top":1791,"left":1721,"width":86,"height":82}],"19":[{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_976_1150_46_63.png","id":70558,"top":1150,"left":976,"width":46,"height":63},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1083_1970_42_55.png","id":70559,"top":1970,"left":1083,"width":42,"height":55},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1334_2351_46_61.png","id":70560,"top":2351,"left":1334,"width":46,"height":61},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1734_1993_43_57.png","id":70561,"top":1993,"left":1734,"width":43,"height":57},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1354_1708_50_54.png","id":70562,"top":1708,"left":1354,"width":50,"height":54}],"37":[{"page":440,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46b_Sadhe_843_1144_72_78.png","id":32009,"top":1144,"left":843,"width":72,"height":78},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Sadhe_1667_2193_83_86.png","id":70663,"top":2193,"left":1667,"width":83,"height":86},{"page":1799,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-49a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_49a_Sadhe_713_1940_92_95.png","id":32007,"top":1940,"left":713,"width":92,"height":95},{"page":2147,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46a_Sadhe_1105_1224_82_82.png","id":70659,"top":1224,"left":1105,"width":82,"height":82},{"page":2147,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46a_Sadhe_1616_1408_82_84.png","id":70660,"top":1408,"left":1616,"width":82,"height":84}]}};

export const formData = {"showBinarized":true,"letterExamples":3,"imageSize":"Small","selectedShelfmarks":["Vat. Syr. 112","Vat. Syr. 140","BL. Add. 17224"],"manuscripts":[],"letters":[{"id":2,"letter":"Alaph (Angular)","is_script":true,"display":"ܐ","script":"estrangela"},{"id":25,"letter":"Lamadh (Final, open)","is_script":true,"display":"ܠ","script":"estrangela"},{"id":37,"letter":"Sadhe","is_script":true,"display":"ܨ","script":"estrangela"}]};

export const columns = [{"property":"letter","visible":true,"props":{"style":{"width":80}}},{"property":"row_remover","props":{"style":{"width":45}},"cell":{"formatters":[null]},"visible":true},{"property":"manuscript1","header":{"label":"Vat. Syr. 112","props":{"label":"Vat. Syr. 112"}},"visible":true,"props":{"msid":9,"style":{"width":200}}},{"property":"manuscript2","header":{"label":"Vat. Syr. 140","props":{"label":"Vat. Syr. 140"}},"visible":true,"props":{"msid":11,"style":{"width":200}}},{"property":"manuscript3","header":{"label":"BL. Add. 17224","props":{"label":"BL. Add. 17224"}},"visible":true,"props":{"msid":16,"style":{"width":200}}}];