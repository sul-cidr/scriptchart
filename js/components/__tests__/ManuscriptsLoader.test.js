/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import ManuscriptsLoader from "../ManuscriptsLoader";
import { defaultManuscripts } from "../__mocks__/appDataMock";

describe("ManuscriptsLoader test", () => {
  it("ManuscriptsLoader should match snapshot", () => {
    const component = renderer.create(
      <ManuscriptsLoader manuscripts={defaultManuscripts} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
