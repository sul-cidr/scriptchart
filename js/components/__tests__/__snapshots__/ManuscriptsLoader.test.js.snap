// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManuscriptsLoader test ManuscriptsLoader should match snapshot 1`] = `
<select
  id="selectedShelfmarks"
  multiple={true}
  name="selectedShelfmarks"
  size="8"
  style={
    Object {
      "fontFamily": "Helvetica Neue, FontAwesome",
    }
  }
  type="string"
>
  <option
    value="Vat. Syr. 092"
  >
    Vat. Syr. 092
     (
    823
    ) 
    
  </option>
  <option
    value="Vat. Syr. 112"
  >
    Vat. Syr. 112
     (
    552
    ) 
    
  </option>
  <option
    value="Vat. Syr. 140"
  >
    Vat. Syr. 140
     (
    528
    ) 
    
  </option>
  <option
    value="BL. Add. 17224"
  >
    BL. Add. 17224
     (
    1173
    ) 
    
  </option>
  <option
    value="BL. Add. 12144"
  >
    BL. Add. 12144
     (
    1081
    ) 
    
  </option>
  <option
    value="BL. Add. 12134"
  >
    BL. Add. 12134
     (
    697
    ) 
    
  </option>
  <option
    value="BL. Add. 14445"
  >
    BL. Add. 14445
     (
    532
    ) 
    
  </option>
  <option
    value="BL. Add. 12146"
  >
    BL. Add. 12146
     (
    1007
    ) 
    
  </option>
  <option
    value="BL. Or. 8732"
  >
    BL. Or. 8732
     (
    770
    ) 
    
  </option>
  <option
    value="BL. Or. 8606"
  >
    BL. Or. 8606
     (
    723
    ) 
    
  </option>
  <option
    value="BL. Add. 17102"
  >
    BL. Add. 17102
     (
    598-599
    ) 
    
  </option>
  <option
    value="BL. Add. 17107"
  >
    BL. Add. 17107
     (
    540-541
    ) 
    
  </option>
  <option
    value="BL. Add. 12175"
  >
    BL. Add. 12175
     (
    533-534
    ) 
    
  </option>
</select>
`;
